docs/quickstart.rst

```diff
@@ -40,9 +40,46 @@ For example, to install or update the latest version using pip:
 .. _PyPI: https://pypi.org/project/Flask-SQLAlchemy/
 
 
+Initialize the Extension
+------------------------
+
+First create the ``db`` object using the ``SQLAlchemy`` constructor.
+
+.. code-block:: python
+
+    from flask import Flask
+    from flask_sqlalchemy import SQLAlchemy
+    from sqlalchemy.orm import DeclarativeBase
+
+    db = SQLAlchemy()
+
+
+By default, this extension assumes that you are using the SQLAlchemy 1.x API for defining models.
+
+To use the new SQLAlchemy 2.x API, pass a subclass of either ``DeclarativeBase`` or ``DeclarativeBaseNoMeta``
+to the constructor.
+
+.. code-block:: python
+
+    from flask import Flask
+    from flask_sqlalchemy import SQLAlchemy
+    from sqlalchemy.orm import DeclarativeBase
+
+    class Base(DeclarativeBase):
+      pass
```

Comment:
Hm, my earlier version of this feature had you call `SQLAlchemy(model_class=DeclarativeBase)` but I wanted to support someone also subclassing MappedAsDataclass, so that's why I now ask developers to create the Base class. They can then choose whether to subclass DeclarativeBase, DeclarativeBaseNoMeta, MappedAsDataclass, etc.

However, I wouldn't typically expect them to put user-defined attributes in Base - those go in the table-specific models.

I'm debating what to write here that doesn't become an essay / repeat SQLAlchemy docs too much.

Reply:
I've added more detail and an additional example, let me know what you think of the change.